File: app/src/main/java/com/sleticalboy/myapplication/StartActivity.java
package com.sleticalboy.myapplication;

import android.app.ListActivity;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.View;
import android.widget.ArrayAdapter;
import android.widget.ListAdapter;
import android.widget.ListView;
import android.widget.TextView;

/**
 * Created on 18-1-29.
 *
 * @author sleticalboy
 * @version 1.0
 * @description
 */
public class StartActivity extends ListActivity {

    private final ItemHolder[] dataList = {
            new ItemHolder(StockActivity.class, "股票 View"),
            new ItemHolder(SmsSenderActivity.class, "加密短信发送"),
    };

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ListAdapter listAdapter = new ArrayAdapter<>(this,
                android.R.layout.simple_list_item_1, dataList);
        setListAdapter(listAdapter);
    }

    @Override
    protected void onListItemClick(ListView l, View v, int position, long id) {
        ItemHolder holder = dataList[position];
        ((TextView) v).setText(holder.mClassName);
        startActivity(new Intent(this, holder.mClass));
    }

    private static class ItemHolder {
        Class<?> mClass;
        String mClassName;

        public ItemHolder(Class<?> aClass, String className) {
            mClass = aClass;
            mClassName = className;
        }

        @Override
        public String toString() {
            return mClassName;
        }
    }
}

File: app/src/main/java/com/sleticalboy/dailywork/ui/activity/StartActivity.kt
package com.sleticalboy.dailywork.ui.activity

import android.app.ListActivity
import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.ArrayAdapter
import android.widget.ListView
import android.widget.TextView

/**
 * Created on 18-1-29.
 *
 * @author sleticalboy
 * @version 1.0
 * @description 启动界面
 */
class StartActivity : ListActivity() {

    private val dataList = arrayOf(
            ItemHolder(LiveCameraActivity::class.java, "TextureView 实现实时预览"),
            ItemHolder(FaceActivity::class.java, "SurfaceView 实现实时预览"),
            ItemHolder(NewCameraActivity::class.java, "camera2 API 测试"),
            ItemHolder(ClassifyActivity::class.java, "RecyclerView 分类别显示"),
            ItemHolder(PagerActivity::class.java, "RecyclerView 分页"),
            ItemHolder(DecorationActivity::class.java, "RecyclerView 添加 item 分割线 / 拖拽排序"),
            ItemHolder(WheelRVActivity::class.java, "RecyclerView 轮播"),
            ItemHolder(PullRefreshActivity::class.java, "下拉刷新库测试"),
            ItemHolder(RefreshActivity::class.java, "下拉刷新 View"),
            ItemHolder(SmsSenderActivity::class.java, "加密短信发送"),
            ItemHolder(StockActivity::class.java, "股票 View")
    )

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val listAdapter = ArrayAdapter(this,
                android.R.layout.simple_list_item_1, dataList)
        setListAdapter(listAdapter)
    }

    override fun onListItemClick(l: ListView, v: View, position: Int, id: Long) {
        val holder = dataList[position]
        (v as TextView).text = holder.mName
        startActivity(Intent(this, holder.mClass))
    }

    class ItemHolder(internal var mClass: Class<*>, internal var mName: String) {

        override fun toString(): String {
            return mName
        }
    }
}
